File: challenges/challenge-samples/php-example-challenge/allTests/app.test.php
<?php
use PHPUnit\Framework\TestCase;
require_once __DIR__ . '/App.php';

/**
 * @testdox longestPalindromicSubstring
 */
class LPSAllTest extends TestCase {

  /** @testdox Returns 'bab' or 'aba' for s='babad' */
  public function test_babad() {
    $res = longestPalindromicSubstring('babad');
    $this->assertContains($res, ['bab', 'aba']);
  }

  /** @testdox Returns 'bb' for s='cbbd' */
  public function test_cbbd() {
    $this->assertSame('bb', longestPalindromicSubstring('cbbd'));
  }

  /** @testdox Returns 'a' for s='a' */
  public function test_a() {
    $this->assertSame('a', longestPalindromicSubstring('a'));
  }

  /** @testdox Returns 'aa' for s='aa' */
  public function test_aa() {
    $this->assertSame('aa', longestPalindromicSubstring('aa'));
  }

  /** @testdox Returns '' for empty string */
  public function test_empty() {
    $this->assertSame('', longestPalindromicSubstring(''));
  }

  /** @testdox Returns 'anana' for s='banana' */
  public function test_banana() {
    $this->assertSame('anana', longestPalindromicSubstring('banana'));
  }

  /** @testdox Returns 'racecar' for s='racecar' */
  public function test_racecar() {
    $this->assertSame('racecar', longestPalindromicSubstring('racecar'));
  }

  /** @testdox Returns 'madam' for s='madam' */
  public function test_madam() {
    $this->assertSame('madam', longestPalindromicSubstring('madam'));
  }

  /** @testdox Returns 'aaaa' for s='aaaa' */
  public function test_aaaa() {
    $this->assertSame('aaaa', longestPalindromicSubstring('aaaa'));
  }
}

File: challenges/challenge-samples/cpp-example-challenge/allTests/app.test.cpp
#include <catch2/catch_test_macros.hpp>
#include "app.hpp"
#include <string>

TEST_CASE("babad -> 'bab' or 'aba'") {
    auto res = longestPalindromicSubstring("babad");
    REQUIRE((res == "bab" || res == "aba"));
}

TEST_CASE("cbbd -> 'bb'") {
    REQUIRE(longestPalindromicSubstring("cbbd") == "bb");
}

TEST_CASE("a -> 'a'") {
    REQUIRE(longestPalindromicSubstring("a") == "a");
}

TEST_CASE("aa -> 'aa'") {
    REQUIRE(longestPalindromicSubstring("aa") == "aa");
}

TEST_CASE("empty -> ''") {
    REQUIRE(longestPalindromicSubstring("") == "");
}

TEST_CASE("banana -> 'anana'") {
    REQUIRE(longestPalindromicSubstring("banana") == "anana");
}

TEST_CASE("racecar -> 'racecar'") {
    REQUIRE(longestPalindromicSubstring("racecar") == "racecar");
}

TEST_CASE("madam -> 'madam'") {
    REQUIRE(longestPalindromicSubstring("madam") == "madam");
}

TEST_CASE("aaaa -> 'aaaa'") {
    REQUIRE(longestPalindromicSubstring("aaaa") == "aaaa");
}
